### DISCUSSION TITLE
Flying EntityCreature

### DISCUSSION THREAD

**Discussion by exfy** [Minestom/Minestom#684] [Help]
I have some questions about flying creatures.
Video: https://www.youtube.com/watch?v=eGsiruIId_w

The flying creature has the following settings:
- getNavigator().getPathingEntity().setAvian(true)

I used the "StrollGoal" with "setAvian(true)", but a Phantom or Blaze always falls to the ground when reaching the target, if the target is lower than the creature. (moveTo has only jump as option?)

How can I give the creature "setNoGravity(true)"?

Also, the creature often **spins extremely** when trying to reach the target: how can I fix this?

Thank you

Best regards
Stefan

**Comment by TheMode** [comment]
> How can I give the creature "setNoGravity(true)"?

You can use `Entity#setNoGravity(true)`, this is likely related to the spinning issue.

**Comment by exfy** [comment]
if i set "entityCreature.getNavigator().getEntity().setNoGravity(true);" to my flying entities, the spinning is much more extreme and the creature almost does not move anymore and the creature still falls to the ground as soon as it reaches a target. So I guess that "setNoGravity(true)" does not work for Phantoms or Blaze?

**Comment by exfy** [comment]
Hello,

does anyone know a solution for this problem, or has encountered similar behavior?
@TheMode 

Thanks for help.
